eng/build-job.yml

```diff
@@ -69,13 +69,9 @@ jobs:
       # Variables used by arcade to gather asset manifests
       - name: _DotNetPublishToBlobFeed
         value: true
-    - name: officialBuildIdArg
```

Comment:
Only thing that sticks out is `officialBuildIdArg` is still used both for `build` and `build-packages` in the yaml. Did we already move the logic to be an unprocessed arg that gets used in the arcade scripts? Or is it still necessary to pass down?

Reply:
I refactored the `officialBuildIdArg` variable to be generated in `xplat-job.yml` so it could be available to multiple jobs.

Reply:
I see the hierarchy, then this makes sense. Thanks 😄